GoogleForCreators/web-stories-wp
# Only updated edited media when exiting edit-mode

## Summary

This removes the immediate update of the data storage of the media item when edited inside the media editor. Instead, edited properties are kept in a local state used for display inside the editor only, and the global store is updated only when the editor is unmounted.

This also removes the reset button inside the media editor, as it was confusing in terms of what it actually resat.

~However, if <kbd>esc</kbd> is pressed, the media editor edits are discarded.~ Not relevant anymore.

## Relevant Technical Choices

* Didn't have to use (and modify) our transform handler at all, as the state can be kept purely internal to the media edit component!

## To-do

* [x] Remove reset button
* [x] Keep edits internal
* [x] Update global store only on unmount
* [x] ~Don't update global store on unmount due to `esc`~ This has been reverted, ignore
* [x] Update and amend tests as required

## User-facing changes

* All edits in media edit-mode is now confined to a single history-entry and can be undone by pressing undo just once

## Testing Instructions

* Add a media element
* Double-click to enter edit mode
* Make multiple separate edits of pan and/or zoom
* Click outside the element or press <kbd>esc</kbd> to exit edit-mode 
* Observe that the media has been updated as desired
* Press undo
* Observe that all the edits are undone in a single undo-step

---



Fixes #1945
